lib/xenops_server.ml

```diff
@@ -2338,6 +2341,17 @@ module VM = struct
          let vm_id = VGPU_DB.vm_of vgpu_id in
          match context.transferred_fd with
          | Some transferred_fd ->
+
+           (* prevent vgpu-migration from pre-Jura to Jura and later *)
+           let _ = try List.assoc key_query_of_sender cookies
```

Comment:
Since the result of `List.assoc` is not used, `List.mem_assoc` could be used to look for the binding in `cookies`.

Reply:
Here we could use a version from the cookie to decide if we can proceed or not, and in the future we could use it to distinguish with new migrations if needed